components/results_table/commons/results_table_base.lua

```diff
@@ -421,6 +423,37 @@ function BaseResultsTable.tournamentDisplayName(placement)
 	return placement.pagename:gsub('_', ' ')
 end
 
+function BaseResultsTable:processVsData(placement)
+	local lastVs
+	if Table.isEmpty(placement.lastvsdata) then
+		lastVs = self:processLegacyVsData(placement)
+	else
+		lastVs = placement.lastvsdata
+	end
+
+	if String.isNotEmpty(lastVs.groupscore) then
+		return placement.groupscore, Abbreviation.make('Grp S.', 'Group Stage')
+	end
+
+	local score = (placement.lastscore or '-') .. SCORE_CONCAT .. (lastVs.score or '-')
+	local vsDisplay = self:opponentDisplay(lastVs, {})
+
+	return score, vsDisplay
+end
+
+-- overwritable
+function BaseResultsTable:processLegacyVsData(placement)
```

Comment:
Imo, have this called as post processing by `queryData()`.

Reply:
hmmm it currently is only called from the results table not from the awards table so would have to test if that breaks
probably later today or tomorrow (work + admining a qualifier atm)

Reply:
calling it in `queryData` seems to be a bad idea as it will kill the option to manally pass data (that has lastvs data in the old style)

would rather call it in `:create()` after the `Table.isEmpty(data)` check